[BUG] Plugin crash when adding "SLAVE" nameserver

**Checklist**
- [x] I have used the latest version of terraform (if not please provide a version number).
- [x] I have used the latest version of this terraform provider (if not please provide a version number).
- [x] I have provided a sample .tf configuration under "Additional context".
- [x] I have answered the following question: Is this issue reproducible in OT&E or only applies to live?

**Describe the bug**
When trying to setup a "SLAVE" type nameserver, the plugin crashes. This crash is reproducible on both live and OT&E.

The nameserver entry at INWX is actually being created, though.

The Log shows the following stack trace:
```
inwx_nameserver.primary: Creating...
╷
│ Error: Plugin did not respond
│
│   with inwx_nameserver.primary,
│   on main.tf line 80, in resource "inwx_nameserver" "primary":
│   80: resource "inwx_nameserver" "primary" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-inwx_v1.4.2 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 98 [running]:
github.com/inwx/terraform-provider-inwx/inwx/internal/resource.resourceNameserverRecordRead({0xd14088, 0xc0001d6a10}, 0xc000507700, {0xb63320?, 0xc0004a6870})
	github.com/inwx/terraform-provider-inwx/inwx/internal/resource/resource_nameserver_record.go:263 +0xd74
github.com/inwx/terraform-provider-inwx/inwx/internal/resource.resourceNameserverCreate({0xd14088, 0xc0001d6a10}, 0xb1a5e0?, {0xb63320?, 0xc0004a6870?})
	github.com/inwx/terraform-provider-inwx/inwx/internal/resource/resource_nameserver.go:246 +0xb6b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000166c40, {0xd13fe0, 0xc00052f470}, 0xd?, {0xb63320, 0xc0004a6870})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:708 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000166c40, {0xd13fe0, 0xc00052f470}, 0xc000552680, 0xc000507580, {0xb63320, 0xc0004a6870})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:838 +0x95a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000365aa0, {0xd14018?, 0xc0004a6b40?}, 0xc0004a6be0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/grpc_provider.go:1021 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002306e0, {0xd13fe0?, 0xc00052ec30?}, 0xc0001d6150)
	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:812 +0x50b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xbc9720?, 0xc0002306e0}, {0xd13fe0, 0xc00052ec30}, 0xc00060c4e0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002b48c0, {0xd17d10, 0xc0003871e0}, 0xc0005326c0, 0xc0002e9ef0, 0x11d6760, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1282 +0xcd5
google.golang.org/grpc.(*Server).handleStream(0xc0002b48c0, {0xd17d10, 0xc0003871e0}, 0xc0005326c0, 0x0)
	google.golang.org/grpc@v1.45.0/server.go:1619 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.45.0/server.go:921 +0x8d
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 26
	google.golang.org/grpc@v1.45.0/server.go:919 +0x246

Error: The terraform-provider-inwx_v1.4.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**To Reproduce**
Run the example Terraform configuration provided.

**Expected behavior**
The DNS zone should be correctly created as a secondary on the INWX nameservers, with the provided configuration on the Terraform resource. The plugin should not crash.

**Screenshots**
~If applicable, add screenshots to help explain your problem.~

**Additional context**

Sample .tf configuration:
```terraform
terraform {
  required_providers {
    inwx = {
      source = "inwx/inwx"
      version = ">= 1.0.0"
    }
  }
}

provider "inwx" {
  api_url = var.api_url
  tan = var.tan
  username = var.username
}

resource "inwx_nameserver" "primary" {
  domain = "test.example"
  type = "SLAVE"
  nameservers = ["ns.inwx.de"]
  master_ip = "192.0.2.1" # RFC5737 example IP
}
```

--- a/inwx/internal/resource/resource_nameserver.go
+++ b/inwx/internal/resource/resource_nameserver.go
@@ -291,8 +291,11 @@ func resourceNameserverRead(ctx context.Context, d *schema.ResourceData, m inter
 		return diags
 	}
 
-	if resData, ok := call["resData"]; ok {
-		resData := resData.(map[string]any)
+	if resDataRaw, ok := call["resData"]; ok {
+		resData, ok := resDataRaw.(map[string]any)
+		if !ok {
+			return diags
+		}
 
 		// Helper function to safely set values
 		setValue := func(key string, field string) {
@@ -308,53 +311,58 @@ func resourceNameserverRead(ctx context.Context, d *schema.ResourceData, m inter
 			}
 		}
 
-		var nsInterface []interface{}
-		// Check if "record" key exists and is an array
-		if records, ok := resData["record"].([]interface{}); ok {
-			for _, item := range records {
-				// Convert item to a map
-				if recordMap, ok := item.(map[string]interface{}); ok {
-					// Check if "type" is "NS" and get "content"
-					// Only include NS records for the root domain (name is empty or equals the domain)
-					if recordType, ok := recordMap["type"].(string); ok && recordType == "NS" {
-						recordName, _ := recordMap["name"].(string)
-						// Only include root NS records, not delegated subzone NS records
-						if recordName == "" || recordName == domain {
-							if content, ok := recordMap["content"].(string); ok {
-								nsInterface = append(nsInterface, content)
+		// SLAVE zones might not have records yet if AXFR hasn't happened.
+		// We skip updating nameservers and soa_mail from records for SLAVE zones
+		// to avoid clearing them if the API returns an empty record list.
+		if d.Get("type").(string) != "SLAVE" {
+			var nsInterface []interface{}
+			// Check if "record" key exists and is an array
+			if records, ok := resData["record"].([]interface{}); ok {
+				for _, item := range records {
+					// Convert item to a map
+					if recordMap, ok := item.(map[string]interface{}); ok {
+						// Check if "type" is "NS" and get "content"
+						// Only include NS records for the root domain (name is empty or equals the domain)
+						if recordType, ok := recordMap["type"].(string); ok && recordType == "NS" {
+							recordName, _ := recordMap["name"].(string)
+							// Only include root NS records, not delegated subzone NS records
+							if recordName == "" || recordName == domain {
+								if content, ok := recordMap["content"].(string); ok {
+									nsInterface = append(nsInterface, content)
+								}
 							}
 						}
-					}
-					// Check if "type" is "SOA" and get "content"
-					if recordType, ok := recordMap["type"].(string); ok && recordType == "SOA" {
-						if content, ok := recordMap["content"].(string); ok {
-							// soa-content split
-							parts := strings.Fields(content)
-							if len(parts) > 1 {
-								// second part is rname
-								rname := parts[1]
+						// Check if "type" is "SOA" and get "content"
+						if recordType, ok := recordMap["type"].(string); ok && recordType == "SOA" {
+							if content, ok := recordMap["content"].(string); ok {
+								// soa-content split
+								parts := strings.Fields(content)
+								if len(parts) > 1 {
+									// second part is rname
+									rname := parts[1]
 
-								soaMail := transformRname(rname)
+									soaMail := transformRname(rname)
 
-								if err := d.Set("soa_mail", soaMail); err != nil {
-									diags = append(diags, diag.Diagnostic{
-										Severity: diag.Error,
-										Summary:  fmt.Sprintf("Could not set %s", rname),
-										Detail:   fmt.Sprintf("Expected %s. %s", rname, err.Error()),
-									})
+									if err := d.Set("soa_mail", soaMail); err != nil {
+										diags = append(diags, diag.Diagnostic{
+											Severity: diag.Error,
+											Summary:  fmt.Sprintf("Could not set %s", rname),
+											Detail:   fmt.Sprintf("Expected %s. %s", rname, err.Error()),
+										})
+									}
 								}
 							}
 						}
 					}
 				}
 			}
-		}
-		if err := d.Set("nameservers", nsInterface); err != nil {
-			diags = append(diags, diag.Diagnostic{
-				Severity: diag.Error,
-				Summary:  fmt.Sprintf("Could not set nameservers %s", err.Error()),
-				Detail:   fmt.Sprintf("Expected %s. %s", nsInterface, err.Error()),
-			})
+			if err := d.Set("nameservers", nsInterface); err != nil {
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Error,
+					Summary:  fmt.Sprintf("Could not set nameservers %s", err.Error()),
+					Detail:   fmt.Sprintf("Expected %s. %s", nsInterface, err.Error()),
+				})
+			}
 		}
 
 		setValue("domain", "domain")

--- a/inwx/internal/resource/resource_nameserver_record.go
+++ b/inwx/internal/resource/resource_nameserver_record.go
@@ -262,10 +262,24 @@ func resourceNameserverRecordRead(ctx context.Context, d *schema.ResourceData, m
 		return diags
 	}
 
-	records := call["resData"].(map[string]any)["record"].([]any)
+	resData, ok := call["resData"].(map[string]any)
+	if !ok {
+		return diags
+	}
+
+	records, ok := resData["record"].([]any)
+	if !ok {
+		// If records are missing, it might be a SLAVE zone that hasn't synced yet.
+		// We don't want to panic. If we were looking for a specific ID and it's not here,
+		// we'll mark it as gone at the end of the function.
+		records = []any{}
+	}
 
 	for _, record := range records {
-		recordt := record.(map[string]any)
+		recordt, ok := record.(map[string]any)
+		if !ok {
+			continue
+		}
 
 		if fmt.Sprintf("%v:%v", d.Get("domain"), recordt["id"]) == d.Id() {
 			d.Set("domain", d.Get("domain").(string))

